### ISSUE TITLE
GTK3 3.22.29, folder panel in the file browser doesn't work anymore.

### ISSUE THREAD

**Issue by TooWaBoo** [Beep6581/RawTherapee#4487]
After updateing to GTK3 3.22.29, the folder panel in the file browser doesn't work anymore.
It only shows the drives. Clicking has no effect.

![grafik](https://user-images.githubusercontent.com/13800920/38378926-7250aeb2-38ff-11e8-80a0-b660edffc6d1.png)

**Comment by heckflosse** [comment]
@TooWaBoo Issue confirmed. In my Win7 it still works partly, but I can't access all folders.

**Comment by gaaned92** [comment]
@TooWaBoo  @heckflosse 

After updating to GTK3 3.22.29-1, here it works as usual.

If you want to test this build:
>RawTherapee_dev_5.4-144-g7328fa92_WinVista_64.zip

uploaded at
https://keybase.pub/gaaned92/RTW64NightlyBuilds/

**Comment by TooWaBoo** [comment]
@gaaned92 
I've tested your build but the folder panel doesn't work. It's the same like my build.

**Comment by heckflosse** [comment]
@TooWaBoo I had to revert glib2 using 
```
pacman -U /var/cache/pacman/pkg/mingw-w64-x86_64-glib2-2.54.3-1-any.pkg.tar.xz
```
to fix the issue.

**Comment by gaaned92** [comment]
I revert also glib2.
If you want avoid update go to /etc/pacman.conf and add
IgnorePkg =mingw-w64-x86_64-glib2

**Comment by heckflosse** [comment]
On my system I had to revert glib2 and gtk3 using

```
pacman -U /var/cache/pacman/pkg/mingw-w64-x86_64-gtk3-3.22.28-1-any.pkg.tar.xz
pacman -U /var/cache/pacman/pkg/mingw-w64-x86_64-glib2-2.54.3-1-any.pkg.tar.xz
```

Just reverting glib2 without reverting gtk3 didn't work (RT didn't start)

**Comment by gaaned92** [comment]
I reverted the two packages and updated the dev build.
I wonder what is the reason why GTK3.22.29 worked on my W10 system and not on @TooWaBoo W10?

**Comment by TooWaBoo** [comment]
Perhaps @Hombre57 should check if this is a bug in GTK3 or something that has to be fixed in RT.

**Comment by TooWaBoo** [comment]
Found this on the web.
https://bugs.archlinux.org/task/57984

https://git.archlinux.org/svntogit/packages.git/commit/trunk?h=packages/gtk3&id=4ac05111e37186cebac1c4fa22610abddb1c2d52

The filechooser has problems too. As far as I understand, it's a problem with the enabled cloudproviders in gtk3 3.22.29 but I've no clue how to fix this.

**Comment by TooWaBoo** [comment]
Could make it work. The culprit is "libgio-2.0-0.dll". I've replaced libgio-2.0-0.dll v2.56.0.0 with libgio-2.0-0.dll v2.54.3.0 (before the GTK3-update) and this fixed it.
(see latest update https://github.com/Beep6581/RawTherapee/issues/4148)

**Comment by gaaned92** [comment]
@TooWaBoo 
perhaps the bug is corrected with new update.

**Comment by TooWaBoo** [comment]
@gaaned92 
No, the bug still exists.

**Comment by gaaned92** [comment]
`No, the bug still exists.`
Yes and it affects all the panels where we want to read or write a file (DCP, curve...)
So stuck to glib2 2.54.3-1 and GTK3 3.22.28-1

Are there some news?
I

**Comment by gaaned92** [comment]
`No, the bug still exists. `

It affects all panels where we want access or read files (DCP, curves..)
I am still stuck on previous versions of glib2 and gtk3
Are there some news?
The archlinux ticket is closed without correction in MSYS2 and I dont feel confident to open a new ticket.

Edit : not longer able to make a working build!
I reverted also Pango to 1.40.11-1

**Comment by Beep6581** [comment]
https://gitlab.gnome.org/GNOME/gtk/issues/170

**Comment by Beep6581** [comment]
Looks like it's fixed in glib-2.56

**Comment by gaaned92** [comment]
mingw-w64-x86_64-glib2 2.56.1-1 is buggy. So waiting for a new version  to get the correction.

Until the correction appears in MSYS2 packages repository, the safe way to make Windows build is to use:
- mingw-w64-x86_64-glib2  2.54.3-1
- mingw-w64-x86_64-gtk3  3.22.28-1
- mingw-w64-x86_64-pango  1.40.11-1
